src/interface/sparse_pack_base.cpp

```diff
@@ -122,12 +123,19 @@ SparsePackBase SparsePackBase::Build(T *pmd, const PackDescriptor &desc) {
   auto coords_h = Kokkos::create_mirror_view(pack.coords_);
 
   // Fill the views
-  ForEachBlock(pmd, [&](int b, mbd_t *pmbd) {
-    int idx = 0;
-    coords_h(b) = pmbd->GetBlockPointer()->coords_device;
+  int idx = 0;
+  ForEachBlock(pmd, [&](int block, mbd_t *pmbd) {
+    int b = 0;
+    if (!desc.flat) {
+      idx = 0;
+      b = block;
+      coords_h(b) = pmbd->GetBlockPointer()->coords_device;
```

Comment:
I guess this makes that no coords object is set for a flat pack since different blocks have different coords. Should we be preventing users from accessing coords from flat packs somehow (at least in debug mode)?

Reply:
Yes that's probably a good idea. That said, it occurs to me we could *can* set the coordinates in a self-consistent way... so maybe we should.

Reply:
done